daemons/ipa-sam/ipa_sam.c

```diff
@@ -2884,6 +2893,23 @@ static bool init_sam_from_td(struct samu *user, struct pdb_trusted_domain *td,
 	}
 	talloc_free(u_sid);
 
+	tmp_ctx= talloc_init("init_sam_from_td");
```

Comment:
Do not create empty context here. Create it off `user` object. It is guaranteed to be allocated with talloc too.